dpnp/dpnp_iface_mathematical.py

```diff
@@ -2106,12 +2103,10 @@ def prod(
 
     """
 
+    dpnp.check_supported_arrays_type(a)
```

Comment:
This can come after the comment and be removed in 2024.1. After removing it, we still have `dpt_array = dpnp.get_usm_ndarray(a)` in the remaining code which will check for supported types.

Reply:
I see, I will move back after the `TODO remove` comment